contracts/modules/governance/NXMaster.sol

```diff
@@ -47,152 +49,169 @@ contract NXMaster is INXMMaster, Governed {
     reentrancyLock = false;
   }
 
-  function upgradeMultipleImplementations(
-    bytes2[] calldata _contractNames,
-    address[] calldata _contractAddresses
+  modifier onlyEmergencyAdmin() {
+    require(msg.sender == emergencyAdmin, "NXMaster: Not emergencyAdmin");
+    _;
+  }
+
+  function addNewInternalContracts(
+    bytes2[] memory newContractCodes,
+    address payable[] memory newAddresses,
+    uint[] memory _types
   )
-  external
+  public
   onlyAuthorizedToGovern
   {
-    require(_contractNames.length == _contractAddresses.length, "Array length should be equal.");
-    for (uint i = 0; i < _contractNames.length; i++) {
-      require(_contractAddresses[i] != address(0), "null address is not allowed.");
-      require(isProxy[_contractNames[i]], "Contract should be proxy.");
-      OwnedUpgradeabilityProxy proxy = OwnedUpgradeabilityProxy(allContractVersions[_contractNames[i]]);
-      proxy.upgradeTo(_contractAddresses[i]);
+    require(newContractCodes.length == newAddresses.length, "NXMaster: Array lengths should be equal.");
+    require(_types.length == newAddresses.length, "NXMaster: Array lengths should be equal.");
```

Comment:
nit: we should use distinct messages, so we know where the error is coming from.